doc/source/tutorial/io.rst

```diff
@@ -5,7 +5,7 @@ File IO (:mod:`scipy.io`)
 
 .. currentmodule:: scipy.io
 
-.. seealso:: `NumPy IO routines <https://www.numpy.org/devdocs/reference/routines.io.html>`__
+.. seealso:: :doc:`NumPy IO routines <numpy:reference/routines.io>`
```

Comment:
This used to go to the [development version of NumPy documentation at numpy.org](https://numpy.org/devdocs/reference/routines.io.html).
Now it goes to [a released version of the documentation at SciPy.org](https://docs.scipy.org/doc/numpy/reference/routines.io.html). 
Do we want that?

Reply:
Interesting! I didn't anticipate that from the entry in objects.inv:
`reference/routines.io                    Input and output                        : reference/routines.io.html`

The relevant configuration seems to be [here](https://github.com/scipy/scipy/blob/master/doc/source/conf.py#L323-L331) in conf.py:
```
intersphinx_mapping = {
    'python': ('https://docs.python.org/dev', None),
    'numpy': ('https://docs.scipy.org/doc/numpy', None),
    'matplotlib': ('https://matplotlib.org', None),
    'asv': ('https://asv.readthedocs.io/en/stable/', None),
}
```

Accordingly, `numpy:` will always resolve to https://docs.scipy.org/doc/numpy/, not https://numpy.org/devdocs/

Reply:
Reverting this for now, as I only intended this PR to make the changes that for sure should be made.  Sorry for the commit spam; I'll be sure to squash at the end.